iam/auth_middleware.go

```diff
@@ -0,0 +1,39 @@
+package iam
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	armoryhttp "github.com/armory-io/go-commons/http"
+	"net/http"
+	"strings"
+)
+
+func AuthMiddleware(ps *ArmoryCloudPrincipalService) func(handler http.Handler) http.Handler {
```

Comment:
This is pretty much exactly taken from the old IAM library: https://github.com/armory-io/lib-go-armory-cloud-iam/blob/main/pkg/core/principal_service.go#L46